### ISSUE TITLE
Fix to #10073 (automatically stringify JSON objects in http requests) introduces hard to diagnose problem

### ISSUE THREAD

**Issue by RoxKilly** [angular/angular#10728]
**I'm submitting a ...**  (check one with "x")
```
[x ] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, instead see https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question
```

**Current behavior**
I have a `screenHttpRequest()` function that is run before every request. Because the API I interact with expects JSON data, this function sets header `content-type: application/json` on every request. This has never been a problem. As of `rc 5` though, probably because of [this commit](https://github.com/angular/angular/commit/83bc5c9), the application will crash with the cryptic error `this._body is null` on `GET` requests. I suspect it's because Angular sees the content-type header and tries to serialize the body, which is of course null on a `GET` request.

**Expected/desired behavior**
A clearer error message there, or an `if(this._body)` check before serialization to avoid the error altogether would be great. 

**Reproduction of the problem**
More details on this [StackOverflow board](https://stackoverflow.com/questions/38908973/unable-to-make-http-request-with-angular-2-rc5/38909123#38909123). The basic steps are

1. set `content-type` header to `application/json`
2. make a request with no body data (eg: an`http.get()` request)

**What is the expected behavior?**
Make the error message more specific or skip the stringifying altogether if the request body is empty

**What is the motivation / use case for changing the behavior?**
With everything going on in a big application, it was tough to isolate the problem because the error is not thrown on the line of code (or even in the function) that sets the header and because the message is so generic (says nothing about http or json).  This can be fixed so easily and at so little cost that it should be addressed.

**Please tell us about your environment:**

* **Angular version:** 2.0.0-rc.5
 
* **Browser:** [ Firefox 48 ] 
 
* **Language:** [ TypeScript ]

**Comment by pkozlowski-opensource** [comment]
This is already fixed in master (7cd4741fcbbea6d58281b3055d1ae7691de1662b). Check #10612 for work-arrounds.

**Comment by RoxKilly** [comment]
thanks
